app/src/main/java/vn/mbm/phimp/me/Camera2.java

```diff
@@ -86,6 +102,73 @@ public Camera2() {
 
 	}
 
+	private boolean inRange(int SubjectValue, int High, int Low) {
+		// In case of a 360
+		SubjectValue =  (360 + (SubjectValue % 360)) % 360;
+		if (High < Low) {
+			return High <= SubjectValue && SubjectValue <= Low;
+		} else {
+			return High <= SubjectValue || SubjectValue <= Low;
+		}
+	}
+
+	private void setFlashIcon() {
+		switch (state) {
+			case 0:
+				flash.setImageResource(R.drawable.flash_on);
+				break;
+			case 1:
+				flash.setImageResource(R.drawable.flash_off);
+				break;
+			case 2:
+				flash.setImageResource(R.drawable.flash_auto);
+				break;
+			default:
+				flash.setImageResource(R.drawable.flash_on);
+				break;
+		}
+	}
+
+	private void rotateIcons(int degrees) {
+		flash.setRotation(degrees);
+		camera_switch.setRotation(degrees);
+		buttonClick.setRotation(degrees);
+	}
+
+	private void cameraFacesNorth() {
+		setFlashIcon();
+		rotateIcons(0);
+	}
+
+	private void cameraFacesEast() {
+		setFlashIcon();
+		rotateIcons(270);
+	}
+
+	private void cameraFacesSouth() {
+		setFlashIcon();
+		rotateIcons(180);
+	}
+
+	private void cameraFacesWest() {
+		setFlashIcon();
+		rotateIcons(90);
+	}
+
+	private void adjustIconPositions() {
```

Comment:
You could have called setFlashIcon from here just before the if stmt instead of repeating it.